src/fromager/sources.py

```diff
@@ -650,7 +650,7 @@ def default_build_sdist(
     #
     # For cases where the PEP 517 approach works, use
     # pep517_build_sdist().
-    sdist_filename = ctx.sdists_builds / (build_dir.name + ".tar.gz")
+    sdist_filename = ctx.sdists_builds / f"{req.name}-{version}.tar.gz"
```

Comment:
After this and #570 both land we'll want to come back here and add the local version, I think.